modules/core/04-channel/keeper/packet.go

```diff
@@ -380,10 +380,10 @@ func (k Keeper) AcknowledgePacket(
 		)
 	}
 
-	if channel.State != types.OPEN {
+	if channel.State != types.OPEN && channel.FlushStatus != types.FLUSHING {
```

Comment:
Don't we want to be able to process acks in `TRYUPGRADE` and `ACKUPGRADE` states?

Reply:
yeah but only if we're flushing, right?

Reply:
aha, yea this is kind of subtle